File: files/CH-124.R
library(tidyverse)
library(readxl)
library(data.table)


path = "files/CH-124 Merge.xlsx"
input = read_excel(path, range = "B2:C7") %>% as.data.table()
input2  = read_excel(path, range = "H2:H9") %>% as.data.table()
test = read_excel(path, range = "I2:I9")  %>% as.data.table()

result = input[input2, on = "Date", roll = "nearest"]

all.equal(result$price, test$Price, check.attributes = FALSE)
#> [1] TRUE

File: files/CH-124.py
import pandas as pd

path = "CH-124 Merge.xlsx"

input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=5).sort_values('Date').reset_index(drop=True)
input2 = pd.read_excel(path, usecols="H:H", skiprows=1, nrows=7) \
    .rename(columns=lambda x: x.replace('.1', '')) \
    .sort_values('Date') \
    .reset_index(drop=True)
test = pd.read_excel(path, usecols="H:I", skiprows=1, nrows=7).rename(columns=lambda x: x.replace('.1', ''))

result = pd.merge_asof(input2, input, on='Date', direction='nearest')
result = pd.merge(result, test, on='Date', how='inner')

print(result["Price"].eq(result["price"]).all()) # True
